std/digest/xxh.d

```diff
@@ -0,0 +1,3183 @@
+    {
+        const size_t nbStripes = ((len - 1) - (block_len * nb_blocks)) / XXH_STRIPE_LEN;
+        assert(nbStripes <= (secretSize / XXH_SECRET_CONSUME_RATE),
+                "nbStripes > (secretSize / XXH_SECRET_CONSUME_RATE)");
+        XXH3_accumulate(acc, input + nb_blocks * block_len, secret, nbStripes, f_acc512);
+
+        /* last stripe */
+        {
+            const ubyte* p = input + len - XXH_STRIPE_LEN;
+            f_acc512(acc, p, secret + secretSize - XXH_STRIPE_LEN - XXH_SECRET_LASTACC_START);
+        }
+    }
+}
+
+enum XXH_SECRET_LASTACC_START = 7; /* not aligned on 8, last secret is different from acc & scrambler */
+
+private ulong XXH3_mix2Accs(const(ulong)* acc, const(ubyte)* secret) @trusted pure nothrow @nogc
+{
+    return XXH3_mul128_fold64(acc[0] ^ XXH_readLE64(secret), acc[1] ^ XXH_readLE64(secret + 8));
+}
+
+private XXH64_hash_t XXH3_mergeAccs(const(ulong)* acc, const(ubyte)* secret, ulong start)
+@trusted pure nothrow @nogc
+{
+    ulong result64 = start;
+    size_t i = 0;
+
+    for (i = 0; i < 4; i++)
+    {
+        result64 += XXH3_mix2Accs(acc + 2 * i, secret + 16 * i);
+    }
+
+    return XXH3_avalanche(result64);
+}
+
+static immutable XXH3_INIT_ACC = [
+        XXH_PRIME32_3, XXH_PRIME64_1, XXH_PRIME64_2, XXH_PRIME64_3,
+        XXH_PRIME64_4, XXH_PRIME32_2, XXH_PRIME64_5, XXH_PRIME32_1
+    ];
+
+private XXH64_hash_t XXH3_hashLong_64b_internal(const(void)* input, size_t len, const(void)* secret,
+        size_t secretSize, XXH3_f_accumulate_512 f_acc512, XXH3_f_scrambleAcc f_scramble) @trusted pure nothrow @nogc
+{
+    align(XXH_ACC_ALIGN) ulong[XXH_ACC_NB] acc = XXH3_INIT_ACC;
+
+    XXH3_hashLong_internal_loop(&acc[0], cast(const(ubyte)*) input, len,
+            cast(const(ubyte)*) secret, secretSize, f_acc512, f_scramble);
+
+    /* converge into final hash */
+    static assert((acc).sizeof == 64, "(acc).sizeof != 64");
+    /* do not align on 8, so that the secret is different from the accumulator */
+    assert(secretSize >= (acc).sizeof + XXH_SECRET_MERGEACCS_START,
+            "secretSize < (acc).sizeof + XXH_SECRET_MERGEACCS_START");
+    return XXH3_mergeAccs(&acc[0], cast(const(ubyte)*) secret + XXH_SECRET_MERGEACCS_START,
+            cast(ulong) len * XXH_PRIME64_1);
+}
+
+enum XXH_SECRET_MERGEACCS_START = 11;
+
+/*
+ * It's important for performance to transmit secret's size (when it's static)
+ * so that the compiler can properly optimize the vectorized loop.
+ * This makes a big performance difference for "medium" keys (<1 KB) when using AVX instruction set.
+ */
+private XXH64_hash_t XXH3_hashLong_64b_withSecret(const(void)* input,
+        size_t len, XXH64_hash_t seed64, const(ubyte)* secret, size_t secretLen) @safe pure nothrow @nogc
+{
+    cast(void) seed64;
+    return XXH3_hashLong_64b_internal(input, len, secret, secretLen,
+            XXH3_accumulate_512, XXH3_scrambleAcc);
+}
+
+/*
+ * It's preferable for performance that XXH3_hashLong is not inlined,
```

Comment:
D has `pragma(inline, false)` for that

Reply:
I know. I will add these performance hacks later. At the moment I care more about working code and corect digest values ;-)

Reply:
I'd suggest to flag this PR as draft.

Reply:
> I'd suggest to flag this PR as draft.

I don't understand what do you mean. The PR is in draft.

Reply:
Applied inline pragmas the same way as for the C code.